### PR TITLE
Update Go to  1.16.2 for the 1.8-dev cycle

### PR DESCRIPTION
**Pull request by sgmiller**
As well as SDK to 1.8-dev.

### CODE REVIEW

**Review by raskchanky — COMMENTED** [review]
I think you have to run `make packages` after changing `packages-oss.yml`, no?

File: .circleci/config/executors/@executors.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ alpine:
 docker-env-go-test-remote-docker:
   resource_class: medium
   docker:
-    - image: "docker.mirror.hashicorp.services/circleci/golang:1.15.4-buster"
```
Comment (swayne275) [inline .circleci/config/executors/@executors.yml:29]: did you do these manually or did it happen automatically?

i seem to remember it not happening automatically, and doing it manually broke tests for me

File: Makefile [hunk 2]
```diff
@@ -17,7 +17,7 @@ EXTERNAL_TOOLS=\
 GOFMT_FILES?=$$(find . -name '*.go' | grep -v pb.go | grep -v vendor)
 
 
-GO_VERSION_MIN=1.15.3
```
Comment (swayne275) [inline Makefile:20]: good find

i'm updating the doc [here](https://docs.google.com/document/d/16sjIk3hzFDPyY5A9ncxTZV_9gnpYSF1_Vx6UA1iiwgI/edit?usp=sharing) too

**Review by swayne275 — APPROVED** [review]

**Review by raskchanky — CHANGES_REQUESTED** [review]
Someone please correct me if I'm wrong, but I think after making changes to `packages-oss.yml`, you need to run `make packages` to update files inside `packages-oss.lock` as well as CI config. When I run `make packages` locally on this branch, there's 4 changed files that aren't included here, so I think that needs to be done before this is ready.

**Comment by sgmiller** [comment]
Thanks, didn't know that.

**Review by briankassouf — APPROVED** [review]

**Review by raskchanky — APPROVED** [review]

**Pull request merged**